### PR TITLE
Add status colours to rows in the status view

### PR DESCRIPTION
**Pull request by huwshimi**

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- start a new model
- add some apps, units, relations and machines
- commit the model
- watch the status view as items change from blue to orange and then to grey

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7990/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7991/

**Comment by frankban** [comment]
s1
In this screenshot, the kibana app is blocked and red, while the units, while blocked, are not highlighted in red.
s2
Similarly, here mysql is in error but that's not reflected in the unit rows.

**Comment by huwshimi** [comment]
I've updated this to display the highest status on the row.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/8002/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/8003/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/8005/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/8006/

File: jujugui/static/gui/src/app/components/status/status.js [hunk 1]
```diff
@@ -6,21 +6,59 @@
...
+      // entity's state property only has a value for pending/error states.
+      return '';
+    }
+    if (!Array.isArray(value)) {
+      value = [value];
+    }
+    const normalised = value.map(val => this._normaliseStatus(val));
+    let status;
+    // Loop through the order of priority until there is a matching status.
+    ['error', 'pending', 'uncommitted', 'ok'].some(val => {
```
Comment (frankban) [inline jujugui/static/gui/src/app/components/status/status.js]: At this point, I'd define these as constants.

**Comment by huwshimi** [comment]
Thanks @frankban and @hatched :shipit:

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/8007/

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**